### ISSUE TITLE
[Twitch] Anon chat does not prevent your username from being registered in a channel's viewer list

### ISSUE THREAD

**Issue by UniversalGeneve1** [night/betterttv#6643]
### Describe the bug

with the "anon chat" toggled on, your username will still appear after the 5 minute refresh, although trying to type into the chat still doesn't work until you run the /join command 

### Steps to reproduce

1) toggle on "anon chat" option in twitch bttv settings
2) check to see of you are not in the user list
3) wait 5 minutes and make sure NOT to type the "/join" command in the chat box
4) check user list, your username will be visible again

### Expected behavior

The expected behavior is that a user with the "anon chat" option will not have their username appear in a channel list they are currently viewing.

### Screenshots

Screenshot showing my username `UniversalGeneve` still appears even with anon chat toggled on
![Screenshot 2024-02-20 164543](https://github.com/night/betterttv/assets/39750103/727dac57-1ae6-44f8-9be2-964fece51656)

### Device information

Windows 10 Home PC

Version      : 22H2
OS Build    : 19045.4046
Experience : Windows Feature Experience Pack 1000.19053.1000.0

google chrome (Version 122.0.6261.58)

### Additional information

Note: I also have the following chrome extensions installed which I believe can manipulate my browser:

7tv
Adblock (NOT Adblock plus)
uBlock Origin

**Comment by night** [comment]
I am not able to reproduce any issues with Anon Chat. This may be caused by 7TV, since it *replaces* Twitch's chat. BetterTTV can only control *native* Twitch chat. It's also worth calling out that user list changes tend to be quite delayed, so after you've left it is not instantly reflected.

**Comment by RalphPuzon** [comment]
@night I was able to get the function to run again by clearing my site cookies. Would you have any insight on why cookies affect this function?

**Comment by night** [comment]
It might be related to a Twitch experiment being ran, which isn't an experiment offered to me.

**Comment by Prasao** [comment]
same problem here

**Comment by alorista** [comment]
I am going through this, as well. I use betterttv and frankerfacez. I was able to immediately disappear or show from the chat list previously. Now, when I click the anon chat slider, nothing happens- even after five minutes, I still show up in the list, and there's no indicator in chat that I'm anonymous (like it previously did) and no 'welcome to the channel' as before. This is happening in chrome and firefox. I have cleared the cache a few times in both browsers, logged in and out of twitch, deleted the addons and re- added them to the browsers. 
Thank you for tracking this. 

*Now that I've written all of that, I'd like to add that, as a last ditch effort, I deleted cookies from the browser one more time from firefox, and now betterttv anon is working just like before!
